diffsims/generators/rotation_list_generators.py

```diff
@@ -159,83 +161,521 @@ def get_grid_around_beam_direction(beam_rotation, resolution, angular_range=(0,
     r = np.ones((psi_theta.shape[0], 1))
     points_in_spherical_polars = np.hstack((r, psi_theta))
-
     # keep only one theta ==0 point, specifically the psi ==0 one
     points_in_spherical_polars = points_in_spherical_polars[
         np.logical_or(
             np.abs(points_in_spherical_polars[:, 2]) > 0,
             points_in_spherical_polars[:, 1] == 0,
         )
     ]
-
     # keep only one theta ==180 point, specifically the psi ==0 one
     points_in_spherical_polars = points_in_spherical_polars[
         np.logical_or(
             np.abs(points_in_spherical_polars[:, 2]) < np.deg2rad(180),
             points_in_spherical_polars[:, 1] == 0,
         )
     ]
-
     points_in_cartesians = vectorised_spherical_polars_to_cartesians(
         points_in_spherical_polars
     )
+    return points_in_cartesians
+
+
+def get_cube_mesh_vertices(resolution, grid_type="spherified_corner"):
+    """
+    Return the (x, y, z) coordinates of the vertices of a cube mesh
+    on a sphere. To generate the mesh, a cube is made to surround the sphere.
+    The surfaces of the cube are subdivided into a grid. The vectors from the
+    origin to these grid points are normalized to unit length. The grid
+    on the cube can be generated in three ways, see grid_type and reference [1].
+
+    Parameters
+    ----------
+    resolution : float
+        The maximum angle in degrees between neighboring grid points. 
+        Where this maximum angle lies depends on the grid_type.
+        To get an integer number of points between the cube face center
+        and the edges, the number of points is rounded up so that
+        resolution is always an upper limit.
+    grid_type : str
+        The type of cube grid, can be either `normalized` or `spherified_edge`
+        or `spherified_corner` (default).
+
+        In the normalized case, the grid on the surface of the cube is linear.
+        The maximum angle between nearest neighbors is found between the <001>
+        directions and the first grid point towards the <011> directions.
+
+        In the spherified_edge case, the grid is constructed so that there
+        are still two sets of perpendicular grid lines parallel to the {100}
+        directions on each cube face, but the spacing of the grid lines is
+        chosen so that the angles between the grid points on the line 
+        connecting the face centers (<001>) to the edges (<011>) are equal.
+        The maximum angle is also between the <001> directions and the first
+        grid point towards the <011> edges.
+
+        The spherified_corner case is similar to the spherified_edge case, but
+        the spacing of the grid lines is chosen so that the angles between
+        the grid points on the line connecting the face centers to the cube
+        corners (<111>) is equal. The maximum angle in this grid is from the
+        corners to the first grid point towards the cube face centers.
```

Comment:
can we juggle this a little? keep the parameter descriptions to the one liners and move everything else into a `Notes`. It's a bit dense at the moment

Reply:
Yep completely agree, just wanted to make sure I wrote everything out